### ISSUE TITLE
Wrong shebang in some python scripts

### ISSUE THREAD

**Issue by acgtyrant** [BVLC/caffe#3458]
I noticed that the shebang of `scripts/download_model_binary.py` is `#!/usr/bin/env python` while it contains a Python 2 library function `urllib.urlretrieve`.

Check these python scripts and fix any wrong shebang please, thank you!

```
scripts/copy_notebook.py
scripts/download_model_binary.py
tools/extra/summarize.py
tools/extra/plot_training_log.py.example
tools/extra/resize_and_crop_images.py
tools/extra/extract_seconds.py
tools/extra/parse_log.py
examples/finetune_flickr_style/assemble_data.py
distribute/python/draw_net.py
distribute/python/detect.py
distribute/python/caffe/classifier.py
distribute/python/caffe/detector.py
distribute/python/classify.py
python/draw_net.py
python/detect.py
python/caffe/classifier.py
python/caffe/detector.py
python/classify.py
```

**Comment by BlGene** [comment]
It would be helpful if you could submit this as a PR.

BR,

**Comment by lukeyeager** [comment]
Caffe is supposed to support Python3 out of the box.

> Caffe’s Python interface works with Python 2.7. Python 3.3+ should work out of the box without protobuf support.
http://caffe.berkeleyvision.org/installation.html#python

So it seems to me the right way to fix this would be to leave the shebang line as-is and add some python version checking like this:
```python
if sys.version_info[0] == 2:
    # use Python2 library
else:
    # use Python3 library
```

**Comment by acgtyrant** [comment]
@lukeyeager You are right, but my employer does not support me to give pull request back to the upstream. So I only reports this issue.

**Comment by shelhamer** [comment]
Fixed by #5181
